operator/v1/types_machineconfiguration.go

```diff
@@ -156,3 +169,116 @@ const (
+type NodeDisruptionPolicyConfig struct {
+	// type represents the type of the object that this policy is in effect for, matching available MachineConfig
+	// fields (files, services, kernel arguments, and sshkeys)
+	// +kubebuilder:validation:Required
+	Type NodeDisruptionPolicyType `json:"type"`
+	// value represents the canoncial name of the object that this policy is in effect for.
+	// This could be a file path, a service name, the specific kernel argument, etc.
+	// +optional
+	Value string `json:"value"`
+	// actions represents the series of commands to be executed on changes to the corresponding type and value
+	// +kubebuilder:validation:Required
+	Actions []NodeDisruptionPolicyAction `json:"actions"`
+}
+
+// Validation is a string enum used in a NodeDisruptionPolicyStatus object. They describe whether the controller
+// was able to validate the latest spec
+// +kubebuilder:validation:Enum:="success";"failed"
+type Validation string
+
+const (
+	// success represents a valid cluster node disruption policy
+	Success Validation = "success"
+
+	// failed represents an invalid cluster node disruption policy
+	Failed Validation = "failed"
+)
+
+// NodeDisruptionPolicyType is a string enum used in a NodeDisruptionPolicyConfig object. They describe the type of
+// the object that this policy will be in effect for.
+// +kubebuilder:validation:Enum:="file";"service";"kernelArgument";"sshKey"
+type NodeDisruptionPolicyType string
+
+const (
+	// File represents a NodeDisruption policy that is in effect for changes to a file.
+	File NodeDisruptionPolicyType = "file"
+
+	// Service represents a NodeDisruption policy that is in effect for changes to a service.
+	Service NodeDisruptionPolicyType = "service"
+
+	// File represents a NodeDisruption policy that is in effect for changes to a kernel argument.
+	KernelArgument NodeDisruptionPolicyType = "kernelArgument"
+
+	// Service represents a NodeDisruption policy that is in effect for changes to a sshKey
+	SSHKey NodeDisruptionPolicyType = "sshKey"
+)
+
+// TODO maybe do this in the MCC validation and skip here?
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'reload' ? has(self.reload) : !has(self.reload)",message="Reload is required when type is reload, and forbidden otherwise"
+// +union
+type NodeDisruptionPolicyAction struct {
+	// type represents the commands that will be carried out if this NodeDisruptionPolicyActionType is executed
+	// Valid value(s): reboot, drain, reload, daemon-reload, none, special
+	// reload requires a corresponding service target specified in the reload field. All other fields cannot have a reload field
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	Type NodeDisruptionPolicyActionType `json:"type"`
+	// reload specifies the service to reload, only valid if type is reload
+	// +optional
+	Reload *string `json:"reload,omitempty"`
```

Comment:
probably want a struct

Reply:
should this be a list so that user can specify multiple servicename together that needs to be reloaded for a NodeDisruptionPolicyType ?

Reply:
done. I opted them to be separate so a user can theoretically order them as they wish to apply in sequence with other actions